### PR TITLE
Added publish script for PowerShell SDK

### PR DESCRIPTION
**Pull request by ismayilov-ismayil**
- Added package nuspec
- Fixed issue with download of archives
- Fixed encoding
- Added publish script

### CODE REVIEW

File: powershell/CompiledHelpers/VstsTaskSdk.csproj [hunk 1]
```diff
@@ -11,9 +11,10 @@
     <AssemblyName>VstsTaskSdk</AssemblyName>
     <SignAssembly>true</SignAssembly>
     <AssemblyOriginatorKeyFile>FinalPublicKey.snk</AssemblyOriginatorKeyFile>
-    <DelaySign>true</DelaySign>    
-    <TargetFrameworkVersion>v4.0</TargetFrameworkVersion>
+    <DelaySign>true</DelaySign>
+    <TargetFrameworkVersion>v4.8</TargetFrameworkVersion>
```
Comment (KonstantinTyukalov) [inline powershell/CompiledHelpers/VstsTaskSdk.csproj:15]: Can this be the same as for an agent?
https://github.com/microsoft/azure-pipelines-agent/blob/4288c014a39b8968228a3f2f85e3da912ae3da79/src/Agent.Service/Windows/AgentService.csproj#L15

Comment (ismayilov-ismayil) [reply powershell/CompiledHelpers/VstsTaskSdk.csproj:15]: The hosted agent which runs this pipeline doesn't contain old msbuild, thus I upgraded it. 
I will revert it back to 4.0 and add old msbuild by task on pipeline.

**Review by KonstantinTyukalov — APPROVED** [review]

**Review by vmapetr — APPROVED** [review]

File: ci/powershell/publish-job.yml
Comment (KonstantinTyukalov) [inline ci/powershell/publish-job.yml:1]: this is not used currently. Will be needed later for CI

**Review by kirill-ivlev — APPROVED** [review]

File: azure-pipelines.yml [hunk 2]
```diff
@@ -22,6 +22,7 @@ jobs:
         nodeVersion: $(nodeVersion)
 
     - template: azure-pipelines-steps-test-build.yml
+    - template: azure-pipelines-steps-nuget-publish.yml
```
[the remark this answers is not in the document]
Comment (KonstantinTyukalov) [reply azure-pipelines.yml]: postponed with CI changes

**Review by tkasparek — APPROVED** [review]

**Pull request merged**